operations/tee.go

```diff
@@ -0,0 +1,178 @@
+package operations
+
+import (
+	"bufio"
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"os/signal"
+	"sort"
+	"sync"
+	"syscall"
+
+	"go.uber.org/zap"
+
+	"go.beyondstorage.io/v5/types"
+)
+
+func (so *SingleOperator) TeeRunViaPipe(path string, expectSize int64) (err error) {
+	ch, err := so.teeViaMultipart(path, expectSize)
+	if err != nil {
+		return err
+	}
+	for v := range ch {
+		if v.Error != nil {
+			return v.Error
+		}
+	}
+
+	fmt.Printf("Stdin is saved to <%s>\n", path)
+	os.Exit(0)
+
+	return err
+}
+
+func (so *SingleOperator) teeViaMultipart(path string, expectSize int64) (errch chan *EmptyResult, err error) {
+	errch = make(chan *EmptyResult, 4)
+	partch := make(chan *PartResult, 4)
+
+	multiparter, ok := so.store.(types.Multiparter)
+	if !ok {
+		return nil, fmt.Errorf("multiparter")
+	}
+
+	mo, err := multiparter.CreateMultipart(path)
+	if err != nil {
+		return nil, err
+	}
+
+	partSize, err := calculatePartSize(so.store, expectSize)
+	if err != nil {
+		return nil, err
+	}
+
+	go func() {
+		// Close partch to inform that all parts have been done.
+		defer close(partch)
+
+		wg := &sync.WaitGroup{}
+		var index int
+
+		r := bufio.NewReader(os.Stdin)
+		b := make([]byte, partSize)
+
+		for {
+			wg.Add(1)
+
+			taskIndex := index
+
+			n, err := io.ReadFull(r, b)
```

Comment:
Do we need to reset `b` here?

Reply:
There is no need to reset `b`. `b` will be overwritten in the course of use.